### PR TITLE
dev-libs/zthread: fix building with clang

### PR DESCRIPTION
**Pull request by austin987**
Gentoo-Bug: https://bugs.gentoo.org/506934

Package-Manager: portage-2.2.26

### CODE REVIEW

**Comment by monsieurp** [comment]
LGMT as far as I can tell.

File: dev-libs/zthread/zthread-2.3.2-r4.ebuild [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+IUSE="debug doc kernel_linux static-libs"
+
+DEPEND="doc? ( app-doc/doxygen )"
+RDEPEND=""
+
+S="${WORKDIR}/${MY_P}"
+
+src_prepare() {
+	rm -f include/zthread/{.Barrier.h.swp,Barrier.h.orig} || die
+	epatch \
```
Comment (SoapGentoo) [inline dev-libs/zthread/zthread-2.3.2-r4.ebuild]: Please turn all `epatch` commands into a bash array `PATCHES`. We want to eliminate epatch in the distant future.

File: dev-libs/zthread/zthread-2.3.2-r4.ebuild [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+		"${FILESDIR}"/${P}-no-fpermissive.diff \
+		"${FILESDIR}"/${P}-m4-quote.patch \
+		"${FILESDIR}"/${P}-automake-r2.patch \
+		"${FILESDIR}"/${P}-gcc47.patch \
+		"${FILESDIR}"/${P}-clang.patch
+	eapply_user
```
Comment (SoapGentoo) [inline dev-libs/zthread/zthread-2.3.2-r4.ebuild]: After having moved everything to `PATCHES`, you just need one call to `default` to cover the `epatch` + `eapply_user` calls

File: dev-libs/zthread/zthread-2.3.2-r4.ebuild [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+		sed -i -e 's|href="html/|href="|' doc/documentation.html || die
+		cp doc/documentation.html doc/html/index.html || die
+		cp doc/{zthread.css,bugs.js} doc/html/ || die
+	fi
+}
+
+src_install() {
+	default
+
+	use doc && dohtml doc/html/*
```
Comment (SoapGentoo) [inline dev-libs/zthread/zthread-2.3.2-r4.ebuild]: `dohtml` is officially deprecated and undesired (https://blogs.gentoo.org/mgorny/2015/11/13/the-ultimate-guide-to-eapi-6/), please use `dodoc` instead.

File: dev-libs/zthread/zthread-2.3.2-r4.ebuild [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+	prune_libtool_files
```
Comment (SoapGentoo) [inline dev-libs/zthread/zthread-2.3.2-r4.ebuild]: The fact that `prune_libtool_files` works is pure coincidence - you need to `inherit eutils` (and the same goes for `epatch`).

File: dev-libs/zthread/files/zthread-2.3.2-clang.patch [hunk 5]
```diff
@@ -0,0 +1,12 @@
+diff -r 2a43e83684d5 src/foreign/zthread/include/zthread/Guard.h
```
Comment (SoapGentoo) [inline dev-libs/zthread/files/zthread-2.3.2-clang.patch]: Also, if possible, could you give a short description in the header of the patch?
https://dev.gentoo.org/~vapier/clean-patches

File: dev-libs/zthread/zthread-2.3.2-r4.ebuild [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+
+	sed -i -e 's:AM_CONFIG_HEADER:AC_CONFIG_HEADERS:' configure.ac || die #467778
+
+	AT_M4DIR="share" eautoreconf
+}
+
+src_configure() {
+	local myconf
+	# Autoconf does not support --disable-debug properly.
+	use debug && myconf="--enable-debug"
```
Comment (SoapGentoo) [inline dev-libs/zthread/zthread-2.3.2-r4.ebuild]: prefer attaching
`$(usex debug "--enable-debug" "")`
to `econf` instead

**Comment by austin987** [comment]
@SoapGentoo thanks for the feedback. Updated.

File: dev-libs/zthread/zthread-2.3.2-r4.ebuild [hunk 7]
```diff
@@ -0,0 +1,71 @@
+# Copyright 1999-2016 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI="6"
+
+inherit eutils autotools
```
Comment (SoapGentoo) [inline dev-libs/zthread/zthread-2.3.2-r4.ebuild]: pet peeve of mine: sort eclasses alphabetically

Comment (austin987) [inline dev-libs/zthread/zthread-2.3.2-r4.ebuild]: I was going to, but saw that -r3 had it that way, so I deferred to previous :). Fixed in next push.

**Comment by austin987** [comment]
@SoapGentoo updated again (use_enable debug & sorted eclasses

**Comment by austin987** [comment]
This was committed here: https://github.com/gentoo/gentoo/commit/5f3562b328dcadeb4f73863bac2128a67df87eda

Thanks for the comments all

**Pull request closed without merging**